src/Templates/MethodTemplate.cshtml

```diff
@@ -227,11 +227,17 @@ func (client @(Model.Owner)) @(Model.Name)(@Model.MethodParametersSignature()) (
 
     @foreach (var p in Model.OptionalHeaderParameters)
     {
+        var deref = string.Empty;
+        if (p.ModelType.IsNumericType())
+        {
+            // optional numeric values are passed as pointers
+            deref = "*";
+        }
         <text>
             if @(p.GetEmptyCheck(p.GetParameterName(), false)) {
             preparer = autorest.DecoratePreparer(preparer,
-            @(string.Format("autorest.WithHeader(\"{0}\",autorest.String({1}))",
-                                      p.SerializedName, p.GetParameterName())))
+            @(string.Format("autorest.WithHeader(\"{0}\",autorest.String({1}{2}))",
+                                      p.SerializedName, deref, p.GetParameterName())))
```

Comment:
I have a question here. I suppose that `autorest.String({1}{2})` here is used for dereference a variable with name of `p.GetParameterName()` and pass it to `autorest.String` method. My question is why this deref symbol `*` comes after the variable name?

Reply:
Sorry, I made a mistake. This is right and the asterisk is before the variable name.